### PR TITLE
Bug: is_minimal_graph with floating nodes

### PR DESCRIPTION
**Pull request by maxelliott-causalens**
## Description and Motivation

- Fixed a bug where metadata on floating nodes would not be correctly carried over to the minimal graph when calling
  `cai_causal_graph.time_series_causal_graph.TimeSeriesCausalGraph.get_minimal_graph`. This also fixes an issue where
  `cai_causal_graph.time_series_causal_graph.TimeSeriesCausalGraph.is_minimal_graph` could return `False` on minimal
  graphs that contain floating nodes.

## How Has This Been Tested?

Unit tests

## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Improvement (non-breaking change which improves functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Refactor (moving code around, general refactoring improvements)
- [ ] Documentation (documentation)

## PR Checklist:

- [X] I have reviewed my own PR? (review the PR as you are reviewing someone else's PR)
- [X] I have implemented all requirements? (see JIRA, project documentation)
- [ ] I am affecting someone else's work? If so: I have added those people as reviewers?
- [ ] I have added comments to all the bits that are hard to follow
- [X] At a bare minimum, I tested this manually
- [X] I have added unit test(s)
- [ ] Is this a bugfix? If so have I added a regression test?
- [X] Does this PR satisfy the [one PR, one concern](https://medium.com/@fagnerbrack/one-pull-request-one-concern-e84a27dfe9f1) rule?
- [ ] If needed, documentation has been added/updated?
- [X] I have updated the appropriate changelog with a line for my changes
- [ ] If this PR breaks reproducibility, have I added a note in the changelog explaining the break in reproducibility
      and its extent?

### CODE REVIEW

**Review by aslan-bakirov — APPROVED** [review]
LGTM!

File: changelog.md [hunk 1]
```diff
@@ -1,5 +1,12 @@
 # Changelog
 
+## NEXT
+
+- Fixed a bug where metadata on floating nodes would not be correctly carried over to the minimal graph when calling
+  `cai_causal_graph.time_series_causal_graph.TimeSeriesCausalGraph.get_minimal_graph`. This also fixes an issue where
+  `cai_causal_graph.time_series_causal_graph.TimeSeriesCausalGraph.is_minimal_graph` could return `False` on minimal
+  graphs that contain floating nodes.
```
Comment (andrew-causalens) [inline changelog.md:8]: By definition is the graph minimal if it has floating nodes? @matteo-causalens what do you think?

File: cai_causal_graph/time_series_causal_graph.py [hunk 2]
```diff
@@ -306,7 +306,14 @@ def get_minimal_graph(self) -> TimeSeriesCausalGraph:
                 if (
                     minimal_cg.variables is None or variable not in minimal_cg.variables
                 ) and not minimal_cg.node_exists(variable):
-                    minimal_cg.add_node(variable)
+                    # Guaranteed there is at least one node, take the first for simplicity as they should be the same
```
Comment (andrew-causalens) [inline cai_causal_graph/time_series_causal_graph.py:309]: Is this true? What if both X and X lag 1 are floating?

Comment (andrew-causalens) [reply cai_causal_graph/time_series_causal_graph.py:309]: I know you test for -1 floating in `test_lagged_floating_nodes_correctly_added_to_minimal_graph ` but what will happen when both are there?

File: changelog.md (hunk 1 above)
Comment (matteo-causalens) [reply changelog.md:8]: yes, we converged to this if I remember well.

File: cai_causal_graph/time_series_causal_graph.py (hunk 2 above)
Comment (andrew-causalens) [reply cai_causal_graph/time_series_causal_graph.py:309]: Extended unit test for this but it fails. How would one suspect this to behave?

Comment (andrew-causalens) [reply cai_causal_graph/time_series_causal_graph.py:309]: Discussed on Slack and taking first is consistent with logic of edges too. Will add backlog ticket but updated unit test for now to be consistent.

**Review by andrew-causalens — APPROVED** [review]

**Pull request merged**